builder/processor/src/main/java/io/helidon/builder/processor/GenerateAbstractBuilder.java

```diff
@@ -0,0 +1,1294 @@
+            pw.println(";");
+            pw.print(SOURCE_SPACING);
+            pw.println("}");
+            pw.println();
+        }
+
+        // setters and getters of builder
+        builderMethods(pw, typeContext);
+
+        toString(pw,
+                 typeContext,
+                 prototype.className() + "Builder",
+                 superType.isPresent(),
+                 typeContext.customMethods().prototypeMethods(),
+                 true);
+
+        // before the builder class is finished, we also generate a protected implementation
+        generatePrototypeImpl(pw, typeContext, typeArgumentNames);
+
+        // end of BuilderBase class
+        pw.print(SOURCE_SPACING);
+        pw.println("}");
+    }
+
+    static void buildRuntimeObjectMethod(PrintWriter pw, TypeContext typeContext, boolean isBuilder) {
+        TypeContext.TypeInformation typeInformation = typeContext.typeInfo();
+        boolean hasRuntimeObject = typeInformation.runtimeObject().isPresent();
+        TypeName builtObject = typeInformation.runtimeObject().orElse(typeInformation.prototype());
+
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("@Override");
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.print("public ");
+        pw.print(builtObject.fqName());
+        pw.println(" build() {");
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.print("return ");
+        if (hasRuntimeObject) {
+            pw.print(builtObject.genericTypeName().fqName());
+            if (isBuilder) {
+                pw.println(".create(this.buildPrototype());");
+            } else {
+                pw.println(".create(this);");
+            }
+        } else {
+            if (isBuilder) {
+                pw.println("return build();");
+            } else {
+                pw.println("return this;");
+            }
+        }
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("}");
+        pw.println();
+
+        // if impl, we also need to add the `get()` method from supplier
+        if (!isBuilder) {
+            pw.print(SOURCE_SPACING);
+            pw.print(SOURCE_SPACING);
+            pw.println("@Override");
+            pw.print(SOURCE_SPACING);
+            pw.print(SOURCE_SPACING);
+            pw.print("public ");
+            pw.print(builtObject.fqName());
+            pw.println(" get() {");
+            pw.print(SOURCE_SPACING);
+            pw.print(SOURCE_SPACING);
+            pw.print(SOURCE_SPACING);
+            pw.println("return build();");
+            pw.print(SOURCE_SPACING);
+            pw.print(SOURCE_SPACING);
+            pw.println("}");
+            pw.println();
+        }
+    }
+
+    static boolean hasConfig(List<PrototypeProperty> properties) {
+        return properties.stream()
+                .filter(it -> "config".equals(it.name()))
```

Comment:
strange to see CONFIG_TYPE here and then a literal "config"

Reply:
it is looking for a method named `config` that returns `CONFIG_TYPE`.

Reply:
In some cases introducing a string constant actually decreases the readability of the code. 
For this case, I am not sure - the `config` method name is used more than once, so maybe a constant could be useful.

Reply:
So the literal `config` is used only once in builder modules, I will keep it as it is